asapdiscovery-data/asapdiscovery/data/tests/test_rdkit.py

```diff
@@ -0,0 +1,449 @@
+import pytest
+from asapdiscovery.data.backend.rdkit import load_sdf
+
+
+def test_ligand_sdf(moonshot_sdf, multipose_ligand, sdf_file):
+    single_conf = load_sdf(moonshot_sdf)
+    assert single_conf.GetNumConformers() == 1
+    multiconf = load_sdf(multipose_ligand)
+
+    assert multiconf.GetNumConformers() == 50
+
+    # this should fail if the file has multiple ligands
+    with pytest.raises(RuntimeError):
+        load_sdf(sdf_file)
+
+
+def test_sd_tag_processing(moonshot_sdf, multipose_ligand):
+    from asapdiscovery.data.backend.rdkit import get_SD_data, set_SD_data
+
+    single_conf = load_sdf(moonshot_sdf)
+    assert get_SD_data(single_conf) == {}
+
+    multiconf = load_sdf(multipose_ligand)
+    assert get_SD_data(multiconf) == {
+        "Compound_ID": [
+            "ADA-UCB-6c2cb422-1",
+            "ADA-UCB-6c2cb422-1",
+            "ADA-UCB-6c2cb422-1",
+            "ADA-UCB-6c2cb422-1",
+            "ADA-UCB-6c2cb422-1",
+            "ADA-UCB-6c2cb422-1",
```

Comment:
This is enormous but I will allow it.

Reply:
Yeah this is terrible. We should probably do an audit of the tests for simplicity and ease. But it works for now, eh?